### PR TITLE
ROX-26151: Pin operator image in operator-bundle (3rd)

### PR DESCRIPTION
**Pull request by mclasmeier**
### Description

This PR changes the existing pipeline for the operator-bundle so that instead of injecting the operator-image tag into the CSV file we resolve this tag to an image digest and then inject that into the CSV.

**Note:** This is a simplified PR: there are no checks about the actual availability of the "replaced" version (think: "previous") of the operator.

### User-facing documentation

- [x] CHANGELOG update is not needed
- [x] documentation PR is not needed

### Testing and quality

- [ ] the change is production ready: the change is GA or otherwise the functionality is gated by a feature flag
- [x] CI results are inspected

#### Automated testing

I have not added any new tests.

### CODE REVIEW

File: operator/bundle_helpers/patch-csv.py [hunk 1]
```diff
@@ -155,7 +155,7 @@ def parse_args():
                         help='Which SemVer version of the operator to set in the patched CSV, e.g. 3.62.0')
     parser.add_argument("--first-version", required=True, metavar='version',
                         help='The first version of the operator that was published')
-    parser.add_argument("--operator-image", required=True, metavar='image',
+    parser.add_argument("--operator-image", metavar='image',
```
Comment (porridge) [inline operator/bundle_helpers/patch-csv.py:158]: Why make this optional? I don't see a place here where we call this script omitting this flag?

**Review by porridge — APPROVED** [review]
Note: I believe this will produce a CSV with an invalid (unsatisfiable) `spec.replaces` of `4.6.0` in the timeframe between 4.7.x tag is created and 4.6.0 is released. This in turn will make the `index-build` and `deploy-previous-via-olm` targets unusable in Konflux.

This may be fine, because we don't plan on doing either automated index building nor e2e testing using the konflux builds (yet). I'm just mentioning this here for awareness.

File: .tekton/determine-image-digest-task.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  - name: determine-image-digest
+    image: registry.access.redhat.com/ubi9/skopeo@sha256:a096332b3a6665b1f1e0b34980406649920cf4babba80aff24a715b76f3f62f3
+    env:
+    - name: REGISTRY_AUTH_FILE
+      value: "$(workspaces.pull-secret.path)/.dockerconfigjson"
+    script: |
+      #!/usr/bin/env bash
+      set -euo pipefail
+      image_ref="docker://$(params.IMAGE_REPOSITORY):$(params.IMAGE_TAG)"
+      digest=$(skopeo inspect \
```
[the remark this answers is not in the document]
Comment (mclasmeier) [reply .tekton/determine-image-digest-task.yaml:31]: done

**Comment by mclasmeier** [comment]
FYI, @porridge pushed two commits.

**Review by porridge — APPROVED** [review]

**Pull request merged**